lms/djangoapps/courseware/access.py

```diff
@@ -801,6 +801,33 @@ def _visible_to_nonstaff_users(descriptor):
     return VisibilityError() if descriptor.visible_to_staff_only else ACCESS_GRANTED
 
 
+def _can_access_descriptor_with_milestones(user, descriptor, course_key):
+    """
+    Returns if the object is blocked by an unfulfilled milestone.
+
+    Args:
+        user: the user trying to access this content
+        descriptor: if not a sequential, access is always granted because only sequentials have milestones
+        course_key: key for the course for this descriptor
+    """
+    # only check milestones for sequentials, as gated content doesn't apply to other block types
+    if descriptor.location.block_type == 'sequential':
+        course_milestones = milestones_helpers.get_course_content_milestones(
+            course_key,
+            descriptor.location,
+            'requires',
+            user.id
+        )
+    else:
+        course_milestones = []
+
+    if unicode(descriptor.location) in [m['content_id'] for m in course_milestones]:
```

Comment:
Once we check for non-sequentials, can remove this `if` statement?  Won't `get_course_content_milestones` return pending milestones only for the given `descriptor.location`?